client/src/js/components/bhRoomSelect/bhRoomSelect.js

```diff
@@ -0,0 +1,48 @@
+angular.module('bhima.components')
+  .component('bhRoomSelect', {
+    templateUrl : 'js/components/bhRoomSelect/bhRoomSelect.html',
+    controller  : RoomSelectController,
+    transclude  : true,
+    bindings    : {
+      uuid      : '<',
+      onSelectCallback : '&',
+      required : '<?',
+      label    : '@?',
+      wardUuid : '<?',
+    },
+  });
+
+RoomSelectController.$inject = ['RoomService', 'NotifyService'];
+
+/**
+ * Room selection component
+ */
+function RoomSelectController(Room, Notify) {
+  const $ctrl = this;
+
+  $ctrl.$onInit = function onInit() {
+    $ctrl.label = $ctrl.label || 'ROOM.TITLE';
+
+    // load all depots
+    Room.read(null, { ward_uuid : $ctrl.wardUuid })
+      .then(rooms => {
+        $ctrl.rooms = rooms;
+      })
+      .catch(Notify.handleError);
+  };
+
+  $ctrl.$onChanges = (changes) => {
+    if (changes.wardUuid) {
+      Room.read(null, { ward_uuid : $ctrl.wardUuid })
```

Comment:
This works which is the most important part.

I am a bit worried that `$ctrl.wardUuid` might not be updated before the `$onChanges` call, and it would use the old value instead of the newly changed value.  In our tests this does happen, but we can make sure that we are definitely using the new value by calling `read()` with the `wardUuid` that came from outside the component.  Like this:

```js
if (changes.wardUuid && changes.wardUuid.currentValue) {
  Room.read(null, { ward_uuid : changes.wardUuid.currentValue })
   // ... etc
}
```